benchmarks/inference/mii/src/plot_repl_scale.py

```diff
@@ -70,36 +62,59 @@ def output_charts(model_size, tp, bs, prompt, gen, log_dir, out_dir):
 
     for c in throughputs:
-
-        # Plotting the scatter plot
-        plt.figure(figsize=(6, 4))
-
-        plt.bar(REPLICA_NUMS, throughputs[c], color="blue", alpha=0.9)
-
-        fit_x_list = np.arange(min(REPLICA_NUMS), max(REPLICA_NUMS), 0.1)
-        mii_fit_model = np.polyfit(REPLICA_NUMS, throughputs[c], 1)
-        mii_model_fn = np.poly1d(mii_fit_model)
-        plt.plot(fit_x_list, mii_model_fn(fit_x_list), color="blue", linestyle="--")
-
-        plt.title(
-            f"Model Llama 2 {model_size.upper()}, Prompt: {prompt}, Generation: {gen}, TP: {tp}"
-        )
-        plt.xlabel("Number of replicas", fontsize=14)
-        plt.ylabel("Throughput (queries/s)", fontsize=14)
-        plt.grid(True)
-        plt.tight_layout()
-        # plt.show()
-        out_file = out_dir / f"repl_scale_llama{model_size}_tp{tp}_p{prompt}g{gen}.png"
-        plt.savefig(out_file)
+        if args.clients_per_replica != None and args.clients_per_replica != c:
+            continue
+        if len(throughputs[c]) == len(replica_nums):
+            print(f"Generating figure for {c} clients/replica.")
+            # Plotting the scatter plot
+            plt.figure()
+
+            plt.bar(replica_nums, throughputs[c], color="blue", alpha=0.9)
+
+            fit_x_list = np.arange(min(replica_nums), max(replica_nums), 0.1)
+            mii_fit_model = np.polyfit(replica_nums, throughputs[c], 1)
+            mii_model_fn = np.poly1d(mii_fit_model)
+            plt.plot(fit_x_list, mii_model_fn(fit_x_list), color="blue", linestyle="--")
+
+            plt.title(
+                f"Model: {model}, Prompt: {prompt}, Generation: {gen}\n\
+                TP: {tp_size}, Clients/Replica: {c}"
+            )
+            plt.xlabel("Number of replicas", fontsize=14)
+            plt.ylabel("Throughput (queries/s)", fontsize=14)
+            plt.grid(True)
+            plt.tight_layout()
+            out_file = out_dir / f"repl_scale_{model}_tp{tp_size}_p{prompt}g{gen}_c_per_r{c}.png"
+            plt.savefig(out_file)
 
 
 if __name__ == "__main__":
-    raise NotImplementedError("This script is not up to date")
     args = get_args()
 
-    for model_size, tps in tp_sizes.items():
-        for tp in tps:
-            for prompt, gen in prompt_gen_pairs:
-                output_charts(
-                    model_size, tp, bs, prompt, gen, args.log_dir, args.out_dir
-                )
+    replica_sets = {}
+
+    result_params = get_result_sets(args)
+
+    # Find all replicas across same sets
+    for model, tp_size, bs, replicas, prompt, gen in result_params:
+        key = f'{model}_{tp_size}_{bs}_{prompt}_{gen}'
+        set_dict = replica_sets.setdefault(f'{model}_{tp_size}_{bs}_{prompt}_{gen}', dict())
+        set_dict.setdefault(f'config', set()).add((model, tp_size, bs, prompt, gen))
+        replicas_list = set_dict.setdefault(f'replicas', list())
+        if replicas not in replicas_list:
+            replicas_list.append(int(replicas))
```

Comment:
```suggestion
    replica_sets = defaultdict(lambda: defaultdict(set))

    result_params = get_result_sets(args)

    # Find all replicas across same sets
    for model, tp_size, bs, replicas, prompt, gen in result_params:
        key = f'{model}_{tp_size}_{bs}_{prompt}_{gen}'
        replica_sets[key]['config'].add((model, tp_size, bs, prompt, gen))
        replica_sets[key]['replicas'].add(int(replicas))
```

Reply:
Thanks for the suggestion, updated relevant scripts.